spec/requirements.md

```diff
@@ -43,3 +47,10 @@ Ideally the entire API could be free of side effects and odd conventions, like m
 
 The system should support exporting generics and classes. If classes are objects, they can be treated like any other object when exporting and importing symbols. If generics are objects, then it should be simple to export all methods on a generic.  It is not clear whether selective method export is important. One use case would be to hide a method with an internal class in its signature to avoid unnecessary documentation. Perhaps `R CMD check` could ignore methods for unexported classes. There should be no need for explicit method registration.  
 
+## Formal arguments
+
+The formal arguments of a generic must be present in every method for that generic. This is in contrast to Julia, where methods can have completely different sets of arguments. We favor a fixed set of formal arguments for the sake of consistency and to enable calling code to depend on a minimal set of arguments that are always present. If the generic formal argument list includes `...`, then methods can add their own arguments. The extra arguments are useful for controlling specialized behaviors, as long as the calling code can assume that calling the generic will always dispatch to a method that supports them. In accordance with the LSP, we could explore enforcing that a method only adds arguments to those of a method dispatching on a parent class. This is easiest to conceptualize and would be most useful in the single dispatch case, but we should also be able to develop a set of rules for nested multiple dispatch.
```

Comment:
"LSP" = ? (spell out in paren).

Not quite clear what in ", as long as the calling code can assume that calling the generic will always dispatch to a method that supports them."    'support' really means. If I think of S3 examples, "support" would rather mean "tolerate", i.e., silently disregard in case the extra arguments used in the call are not part of that method's arguments .. for which then, that method (and many others?) may need to also get the `...`  as Hadley mentioned somewhere earlier in the #7